### PR TITLE
iOSDeviceManager partially implements PhysicalDevice::LifeCycle interface

### PR DESCRIPTION
**Pull request by jmoody**
### Motivation

Progress on:

* Calabash iOS: Created focused test for daily deploy [VSTS](https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems/edit/36207)

We need a local solution for clear-app-data.

### CODE REVIEW

File: lib/run_loop/physical_device/ios_device_manager.rb [hunk 1]
```diff
@@ -22,67 +33,178 @@ def initialize(device)
...
+        )
+        hash[:out]
+      end
+
+      def can_reset_app_sandbox?
+        true
+      end
+
+      def reset_app_sandbox(app_or_ipa)
+        xcappdata = empty_xcappdata_path
```
Comment (polyakrecords) [inline lib/run_loop/physical_device/ios_device_manager.rb]: Let's change this method using new iOSDeviceManager command – `clear-app-data`?

File: lib/run_loop/physical_device/ios_device_manager.rb [hunk 2]
```diff
@@ -22,67 +33,178 @@ def initialize(device)
...
 
-      def uninstall_app(bundle_id)
-        return true if !app_installed?(bundle_id)
+=begin
+Private Methods
+=end
+
+      private
+
+      def empty_xcappdata_path
```
Comment (polyakrecords) [inline lib/run_loop/physical_device/ios_device_manager.rb]: Do we need this if we'll use iOSDeviceManager `clear-app-data` command?

**Comment by jmoody** [comment]
Don't merge yet.  I need to do some follow-up testing.

**Comment by jmoody** [comment]
Tests are green for Xcode 8.3.3 - Xcode 9.4 beta vs compatible iOS physical devices.

We need to wait for a iOSDeviceManager change before merging.

**Pull request merged**